descartes_light/core/include/descartes_light/core/waypoint_sampler.h

```diff
@@ -59,6 +59,14 @@ class WaypointSampler
    * sampling step
    */
   virtual typename std::vector<StateSample<FloatType>> sample() const = 0;
+
+  virtual std::ostream& format(std::ostream& out) const
+  {
+    out << "WaypointSampler format() not defined";
+    return out;
+  }
+
+  friend std::ostream& operator<<(std::ostream& os, const WaypointSampler<FloatType>& wps) { return wps.format(os); }
```

Comment:
```suggestion
  friend std::ostream& operator<<(std::ostream& os, const WaypointSampler<FloatType>& wps) { return os; }
```

The `format` method seems unnecessary. Let's replace it by just letting the classes customize the `ostream` operator